app/src/main/java/clarkson/ee408/tictactoev4/PairingActivity.java

```diff
@@ -103,23 +166,80 @@ public void updateAvailableUsers(List<User> availableUsers) {
      * @param userOpponent the User to send invitation to
      */
     private void sendGameInvitation(User userOpponent) {
-        // TODO:  Send an SEND_INVITATION request to the server. If SUCCESS Toast a success message. Else, Toast the error
+        //Create request object with type SEND_INVITATION
+        Request request = new Request();
+        request.setType(RequestType.SEND_INVITATION);
+
+        //SEND_INVITATION request if successful, Toast success or error
+        AppExecutors.getInstance().networkIO().execute(() -> {
+            try {
+                Response ir = SocketClient.getInstance().sendRequest(request, InvitationResponse.class);
```

Comment:
There is no "InvitationResponse" class.

```suggestion
                Response ir = SocketClient.getInstance().sendRequest(request, Response.class);
```